src/prognostic_equations/eddy_diffusion_closures.jl

```diff
@@ -527,6 +530,90 @@ function mixing_length(
     return MixingLength{FT}(l_final, l_W, l_TKE, l_N, l_grid)
 end
 
+function mixing_length(
+    params,
+    ustar,
+    ᶜz,
+    z_sfc,
+    ᶜdz,
+    sfc_tke,
+    ᶜN²_eff,
+    ᶜtke,
+    obukhov_length,
+    ᶜstrain_rate_norm,
+    ᶜPr,
+    ᶜtke_exch,
+    scale_blending_method,
+    property::Val{P},
+) where {P}
+
+    ᶜmixing_length_tuple = mixing_length_lopez_gomez_2020(
+        params,
+        ustar,
+        ᶜz,
+        z_sfc,
+        ᶜdz,
+        sfc_tke,
+        ᶜN²_eff,
+        ᶜtke,
+        obukhov_length,
+        ᶜstrain_rate_norm,
+        ᶜPr,
+        ᶜtke_exch,
+        scale_blending_method,
+    )
+
+    return getproperty(ᶜmixing_length_tuple, P)
+end
+
+function mixing_length(Y, p, mixing_length_property = :master)
+    (; params) = p
+    (; ustar, obukhov_length) = p.precomputed.sfc_conditions
+    (; ᶜtke⁰) = p.precomputed
+    (; ᶜlinear_buoygrad, ᶜstrain_rate_norm) = p.precomputed
+    ᶜz = Fields.coordinate_field(Y.c).z
+    z_sfc = Fields.level(Fields.coordinate_field(Y.f).z, Fields.half)
+    ᶜdz = Fields.Δz_field(axes(Y.c))
+    sfc_tke = Fields.level(ᶜtke⁰, 1)
+
+    ᶜprandtl_nvec_lazy = turbulent_prandtl_number(p)
+    ᶜprandtl_nvec = p.scratch.ᶜtemp_scalar_7
+    ᶜprandtl_nvec = Base.Broadcast.materialize(ᶜprandtl_nvec_lazy)
```

Comment:
Can this also be done lazily?

Reply:
Making it lazy leads to the ` Kernel invocation uses too much parameter memory` error on the GPU.